quantum/bits.h

```diff
@@ -0,0 +1,23 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+
+#pragma once
+
+#define BITS_PER_BYTE 8
+#define BITS_PER_LONG 32
+#define BITS_PER_LONG_LONG 64
+
+#define BIT(nr) (UL(1) << (nr))
+#define BIT_ULL(nr) (ULL(1) << (nr))
+#define BIT_MASK(nr) (UL(1) << ((nr) % BITS_PER_LONG))
+#define BIT_WORD(nr) ((nr) / BITS_PER_LONG)
```

Comment:
I think the intended purpose of `BIT_MASK()` and `BIT_WORD()` is to select the nth bit from a uint32_t array, is that right? Are there other notable use cases besides this? Like:

```c
uint32_t array[N];
// Get nth bit from `array`.
array[BIT_WORD(n)] & BIT_MASK(n)
```

I have to say, I find this unclear. The notion of a "word" is prone to misunderstanding, since it depends on context what the word data unit is. It seems easy to overlook a mismatch between the array datatype and the word size supposed by these macros. I suggest again that the bit width be stated explicitly. Arguably, we could just as well reuse the `BIT32()` macro [né `BIT()`] from above, eliminating the `_MASK` and `_WORD` macros. Like:

```c
// Get nth bit from `array`.
array[n / 32] & BIT32(n % 32) 
```

Would this be an improvement?

Reply:
To be honest I haven't used `BIT_MASK()` and `BIT_WORD()` myself but that seems to be the intended use-case. In the current iteration they have been removed as I don't see any benefit from those just like you.